app/controllers/vendor_api/notes_controller.rb

```diff
@@ -0,0 +1,36 @@
+module VendorAPI
+  class NotesController < VendorAPIController
+    before_action :validate_metadata!
+
+    include ApplicationDataConcerns
+
+    def create
+      @new_note_form = AllowedCrossNamespaceUsage::ProviderInterfaceNewNoteForm.new(note_params)
+
+      if @new_note_form.save
+        application_choice = application_choices_visible_to_provider.find(params[:application_id])
+
+        render json: %({"data":#{ApplicationPresenter.new(version_number, application_choice).serialized_json}})
+      else
+        render_validation_errors(@new_note_form.errors)
+      end
+    end
+
+    def render_validation_errors(errors)
+      error_responses = errors.full_messages.map { |message| { error: 'ValidationError', message: message } }
+      render status: :unprocessable_entity, json: { errors: error_responses }
+    end
+
+    def note_params
+      {
+        application_choice: application_choice,
+        user: audit_user,
+        message: params[:data][:message],
+      }
+    end
+
+    def application_choice
+      @application_choice ||= application_choices_visible_to_provider.find(params[:application_id])
+    end
```

Comment:
this is no longer needed

https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/controllers/concerns/vendor_api/application_data_concerns.rb#L9
```suggestion
 
```